internal/kafka.go

```diff
@@ -168,12 +168,20 @@ func handleCommand(
 		}
 		break
 	case "challenge_session":
+		if command.SessionId == "" {
+			log.Printf("KAFKA: challenge_session: session_id is EMPTY, break\n")
+			break
+		}
 		// exempt a site from challenge according to config
 		_, disabled := config.SitesToDisableBaskerville[command.Host]
 
 		if !disabled {
 			// gin does urldecode or cookie, so we decode any possible urlencoded session id from kafka
-			sessionIdDecoded, _ := url.QueryUnescape(command.SessionId)
+			sessionIdDecoded, decodeErr := url.QueryUnescape(command.SessionId)
```

Comment:
Do you think it's worth it to add a security check of the decoded value before using it as a session id? I'm thinking in a scenario where the cookie is modified intentionally to break or overload the service.